### PR TITLE
Catch-up with `main`.

### LINKED ISSUE #2850: Merge main to release-v2.0
main has diverged from the point at which release-v2.0 branched. To avoid regressions from v1.3.0 to v2.0.0, we need to merge main to release-v2.0. The main source of conflicts will be #2734, once it lands.

/cc @mapbox/navigation-ios @avi-c

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

Catch-up `release-v2.0` with `main`.

Fixes #2850.

### CODE REVIEW

**Comment by MaximAlien** [comment]
There is a chance that https://github.com/mapbox/mapbox-navigation-ios/pull/2881 wasn't merged as expected and fix for https://github.com/mapbox/mapbox-navigation-ios/issues/2766 might not fully work. @azarovalex, it'd be great if you could affirm this.

File: Example/AppDelegate.swift [hunk 1]
```diff
@@ -12,8 +13,20 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
 
+    @available(iOS 12.0, *)
+    lazy var interfaceController: CPInterfaceController? = nil
+
+    @available(iOS 12.0, *)
+    lazy var carWindow: CPWindow? = nil
+
+    @available(iOS 12.0, *)
+    lazy var sessionConfiguration: CPSessionConfiguration? = nil
+
```
Comment (MaximAlien) [inline Example/AppDelegate.swift:24]: @avi-c, do we really need these properties in `AppDelegate.swift`?

Comment (1ec5) [reply Example/AppDelegate.swift:24]: Good catch, these properties appear to be unused, if I’m not mistaken.

**Comment by 1ec5** [comment]
For reference, this file resolves conflicts in the files listed in the commit message for 4128beece06d483f16a5e5dfca21b3fa9e3dc414.

**Review by 1ec5 — APPROVED** [review]

File: Sources/MapboxNavigation/NavigationViewController.swift [hunk 2]
```diff
@@ -42,8 +42,6 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
         }
     }
     
-    var didChangeAuthorizationIsFirstCalled = true
-
```
Comment (jill-cardamon) [inline Sources/MapboxNavigation/NavigationViewController.swift]: The delegate method `locationManagerDidChangeAuthorization(_:)` is called at startup and whenever the user changes authorization permissions. `didChangeAuthorizationIsFirstCalled` was to deal with the first case. Without this property, the halo view and enable precise location banner would show up when they weren't supposed to.

Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationViewController.swift]: Thanks for flagging that. I've tried on several tests and do not see any issues, at the same time to bring back your implementation from `main` we'll have to possibly bring back `MBNavigationAccuracyAuthorization` and maybe other APIs. I'm inclined to merge this PR as is and create another ticket in scope of which we'll perform more through testing and add fixes if needed.

**Pull request merged**